### PR TITLE
HW#2 - Oleg Ivantsov

### PR DESCRIPTION
**Pull request by IvantsovOleg**

### CODE REVIEW

File: src/scss/flexbox.scss [hunk 1]
```diff
@@ -0,0 +1,196 @@
+html, body {
```
Comment (nvkuznetsova) [inline src/scss/flexbox.scss:1]: Please put these styles after all `@font-face` rules. Don't forget `@font-face` rules should be before all other styles for better performance

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    <meta charset="UTF-8">
+    <title>New_Jersey - HW#2</title>
+    <link rel="stylesheet" href="css/flexbox.css">
+</head>
+<body style="background-color: #505a62">
+<div class="weather-widget">
+    <div class="weather-widget__row weather-widget__row--refresh">
+        <img alt="refr" src="img/refresh.svg"/>
+    </div>
+    <div class="weather-widget__row weather-widget__row--cloud">
```
Comment (nvkuznetsova) [inline public/index.html]: `weather-widget__row--cloud` is incorrect modifier name. A modifier describes the behavior or appearance of a block or element. Or maybe you can use a mix (https://ru.bem.info/methodology/quick-start/#%D0%BC%D0%B8%D0%BA%D1%81).
I'm not very experienced in BEM naming but I think `weather-widget__row--theme-cloud` will suit more.

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        <img alt="cloud" src="img/cloud.svg"/>
+    </div>
+    <div class="weather-widget__row widget-data">
+        <div class="widget-data__degrees">79&deg;</div>
+        <div class="widget-data__location">
+            <div class="widget-data__location--appearance">Cloudy Skies</div>
```
Comment (nvkuznetsova) [inline public/index.html]: The modifier CSS class should always be used paired with the class of its block or element. But here you can just convert the modifier class into element class `widget-data__appearance`. 
Same for `widget-data__location--city`.

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            <div class="widget-data__location--city">Sicklerville, New Jersey</div>
+        </div>
+        <div class="widget-data__date">
+            <div>MAY</div>
+            <div>21</div>
+        </div>
+    </div>
+    <div class="weather-widget__row weather-widget__row--empty-space"></div>
+    <div class="weather-widget__row weather-widget__row--footer">
```
Comment (nvkuznetsova) [inline public/index.html]: Incorrect modifier name. `weather-widget__row--direction-bottom` would suit more.

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        <div class="additional-information">
+            <div class="additional-information__wrapper">
+                <div class="additional-information__wrapper--image">
```
Comment (nvkuznetsova) [inline public/index.html]: Convert modifier class to element class. Same for `additional-information__wrapper--info`

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            <a href="#"><img alt="0" src="img/pagination-0.svg"/></a>
+        </div>
+    </div>
+
+
+
+
+
+
+<!--    <div class="weather-widget__refresh">-->
```
Comment (nvkuznetsova) [inline public/index.html]: Seems this part won't be necessary anymore) Please remove.

File: public/index.html [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        <div class="widget-data__location">
+            <div class="widget-data__location--appearance">Cloudy Skies</div>
+            <div class="widget-data__location--city">Sicklerville, New Jersey</div>
+        </div>
+        <div class="widget-data__date">
+            <div>MAY</div>
+            <div>21</div>
+        </div>
+    </div>
+    <div class="weather-widget__row weather-widget__row--empty-space"></div>
```
Comment (nvkuznetsova) [inline public/index.html]: No need to use a separate element to define the distance between elements. Just use `margin`)

File: src/scss/flexbox.scss [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+
+    &__row {
+        width: $width;
+        text-align: center;
+
+        &--refresh {
+            flex: 0 1 10.9%;
```
Comment (nvkuznetsova) [inline src/scss/flexbox.scss:87]: I don't mind using CSS shorthand property if you experienced with flex. But for training, I'd recommend using separate properties for each value next time. No need to fix this)

**Comment by nvkuznetsova** [comment]
Your widget looks nice in all browsers but it is not responsive. 
This behavior is not responsive.
not-responsive

Here the example of the responsive widget.
responsive

Try to add `media-queries` and use `min-width` and `max-width` propesties instead of exact `width` value.

Moreover, You need 2 styles options using `grid` and `flexbox` as it is described in the task
image

Please add a file with styles using `grid`. And please don't forget to compile `grid` sass styles to CSS, so it was easy to switch between solutions.

**Comment by IvantsovOleg** [comment]
I added the grid layout where used `media-queries` in it. I admit that I have problems with UX and UI design... so it is difficult for me to imagine how the widget should look on different screens, so my imagination "born" only one additional option for the layout of elements inside the widget, I'm sorry )))

**Pull request closed without merging**